app/models/user.rb

```diff
@@ -560,8 +560,12 @@ def admin?
     Rails.env.production? && EnvConfig.WCA_LIVE_SITE? ? software_team_admin? : software_team?
   end
 
+  def delegate_roles
+    active_roles.filter { |role| UserRole.is_group_type?(role, UserGroup.group_types[:delegate_regions]) }
+  end
+
   def any_kind_of_delegate?
-    delegate_status.present? || active_roles.any? { |role| UserRole.is_group_type?(role, UserGroup.group_types[:delegate_regions]) }
+    active_roles.any? { |role| UserRole.is_group_type?(role, UserGroup.group_types[:delegate_regions]) }
```

Comment:
It is my understanding that `active_roles` is basically just a filtered version of `has_many :roles, class_name: "UserRole"`, i.e. `active_roles` can only ever hold "real" roles from the new role system.

This would mean that with your proposed changes of this PR, some Delegates who don't have a "proper" role yet will return `false` here when they are supposed to return `true`.

In other words, are we confident that each and every single Delegate in our whole database already has a new `UserRole` mapping?

Reply:
Currently `has_many :roles` is getting overriden by `def roles`. So it can hold `unreal` roles as well. So this will work as expected.